src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers/TypeUpgradeAnalyzer.cs

```diff
@@ -133,14 +133,28 @@ private static void AnalyzeIdentifier(SyntaxNodeAnalysisContext context, IEnumer
                 return;
             }
 
-            // If the identifier resolves to an actual symbol that isn't the old identifier, bail out
-            // Also refrain from adding a diagnostic if the symbol is ambiguous but a candidate symbol matches the
-            // new type. This is useful in cases where the type is referenced as a simple name and the new namespace
-            // has been added but the old namespace hasn't been removed yet.
+            var fullyQualifiedNameNode = context.Node.GetQualifiedName();
+
+            // If the identifier is part of a fully qualified name and the qualified name does not match
+            // the old name, then bail out because it is not referring to that type
+            if ((fullyQualifiedNameNode is CSSyntax.QualifiedNameSyntax || fullyQualifiedNameNode is VBSyntax.QualifiedNameSyntax)
```

Comment:
this would be nice to have in a separate file (whenever I see string comparisons like this, I'm betting we'll do it multiple times somewhere and it would be nice to hide it behind some logical method)

Reply:
Agreed. I'll make an extension method in SyntaxNodeExtensions.Predicates for this.

Reply:
Hah - just went to make the extension method and found it was already there! I'll start using the existing extension method :)